tag_hierarchy/unittests/test_is_suppressed_field.cpp

```diff
@@ -0,0 +1,135 @@
+}
+
+// Verify that the suppression relationship is correct for input filter.
+void verify_correct_suppression_relationship_with_filter(const std::vector<NodeType>& filter) {
+    // Construct the query for the top nodes.
+    auto query = std::vector<NodeType>(
+            {{
+                {std::string("command"), std::string("nodes")},
+                {std::string("parentId"), pybind11::none()}
+             }}
+    );
+    for (const auto& filterEntry : filter) {
+        query[0].insert(filterEntry.begin(), filterEntry.end());
+    }
+
+    auto response = TagHierarchy::Handle(query);
+
+    // Construct the command to pass on when querying for child nodes.
+    auto childNodesCommand = std::vector<NodeType>(
+            {{
+                     {std::string("command"), std::string("nodes")}
+             }}
+    );
+
+    // Insert filter into the command.
+    for (const auto& filterEntry : filter) {
+        childNodesCommand[0].insert(filterEntry.begin(), filterEntry.end());
+    }
+
+    // Verify the suppression relationship for all nodes in the query result.
+    for (const auto& props : response) {
+        std::string nodeId = boost::get<std::string>(props.find("id")->second);
+        verify_correct_suppression_relationship(
+                boost::get<bool>(props.find("issuppressed")->second),
+                nodeId,
+                childNodesCommand
+        );
+    }
+}
+
+BOOST_FIXTURE_TEST_SUITE( FieldIsSuppressedTest, Fixture );
+
+    // Tests that the field for alarm suppression is propagated from
+    // model element nodes all the way up to the root node.
+    BOOST_AUTO_TEST_CASE( test_is_suppressed_propagation )
+    {
+        verify_correct_suppression_relationship_with_filter(std::vector<NodeType>());
+    }
+
+    // Tests that the field for alarm suppression is propagated from
+    // model element nodes all the way up to the root node, when the query
+    // contains a model owner filter.
+    BOOST_AUTO_TEST_CASE( test_is_suppressed_propagation_with_model_owner_filter )
+    {
+        verify_correct_suppression_relationship_with_filter(std::vector<NodeType>(
+                {{
+                    {
+                        std::string("modelownerfilter"), std::vector<std::string>{"8eb4a1e1-1316-bcef-b20f-bbbaed2a4e95"}
+                    }
+                }}
+                ));
+    }
+
+    // Tests that the field for alarm suppression is propagated from
+    // model element nodes all the way up to the root node, when the query
+    // contains a model class filter.
+    BOOST_AUTO_TEST_CASE( test_is_suppressed_propagation_with_model_class_filter )
+    {
+        verify_correct_suppression_relationship_with_filter(std::vector<NodeType>(
+                {{
+                         {
+                             std::string("modelclassfilter"), std::vector<std::string>{"44380be8-76d2-a1fd-dc50-0ad8f4c151e4"}
+                         }
+                }}
+                ));
+    }
+
```

Comment:
Great job with the testing!
A few camelCases around here, also in the other files.

Reply:
Changed all local variables to snake case.